src/nspawn/nspawn.c

```diff
@@ -378,6 +379,22 @@ static void parse_share_ns_env(const char *name, unsigned long ns_flag) {
         arg_clone_ns_flags = (arg_clone_ns_flags & ~ns_flag) | (r > 0 ? 0 : ns_flag);
 }
 
+
+static void parse_mount_settings_env(void) {
+        const char *e;
+
+        e = getenv("SYSTEMD_NSPAWN_API_VFS_WRITABLE");
+        if (!e)
+                return;
+
+        if (streq(e, "yes"))
+                arg_mount_settings &= ~MOUNT_MASK_RO;
+        else if (streq(e, "no"))
+                arg_mount_settings |= MOUNT_MASK_RO;
+        else if (streq(e, "network"))
+                arg_mount_settings |= MOUNT_MASK_NETNS;
```

Comment:
So, I think we should use the "extended bool" logic to parse this, which we frequently use in other cases like this. Specifically, this means checking first if the parameter has the "network" value. And if not, then fall back to parsing it as a boolean via parse_boolean(). That way, people can treat it as a usual boolean (and also use "0", "off" and "false" as negative value, and "1", "on", "true" as positive value), except that it understands one more, special value, which is "network".

I also think this should generate a warning message, when we don't understand the passed parameter. i.e. use log_warning() and say that the value set in the env var is not understood and will be ignored.